pkgs/applications/graphics/potreeconverter/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, boost
+, tbb
+}:
+
+stdenv.mkDerivation rec {
+  pname = "PotreeConverter";
+  version = "unstable-2022-08-04";
+
+  src = fetchFromGitHub {
+    owner = "potree";
+    repo = "PotreeConverter";
+    rev = "758bbac98a662de5e57d2280675e11cc76241688";
+    sha256 = "sha256-pDdV2/edYhhBWs153hSy1evI3cXD0Xq9nrEsw3JNcH4=";
+  };
+
+  buildInputs = [
+    boost
+    tbb
+  ];
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  patchPhase = ''
+    substituteInPlace ./CMakeLists.txt \
+      --replace "find_package(TBB REQUIRED)" ""
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/lib
+    mv liblaszip.so $out/lib
+    mv PotreeConverter $out/bin
```

Comment:
This can be done with `install(2)`: `-Dm755` for executables and `-Dm655` for everything else. Also, please add `preInstall` and `postInstall` hooks.

Reply:
One of the primary purposes of the `/nix/store` is that we don't need to care about permissions anymore. By putting it in `/bin`, I thought it was already marked as executable. What is the benefit of using `install -Dm755`? I thought permission bits other than 'execute' don't mean anything, as all the Nix store knows about is `execute`. Even if you set it to `755` or `655`, Nix will normalise it to either `0555` or `0444` according to the thesis.

See section `5.2 The Nix store` of https://edolstra.github.io/pubs/phd-thesis.pdf.

> Also note that we do not keep any meta-information about files other than the executable bit

And on page 112

> The permission on each file is set to octal 0444 or 0555, meaning that the file
is readable by everyone and writable by nobody, but possibly has execute per-
mission. Removing write permission prevents the path from being accidentally
modified when it is used as an input to other derivations.

Reply:
> By putting it in `/bin`, I thought it was already marked as executable.

Putting it in bin doesn't magically mean it will get marked as executable.


> I thought permission bits other than 'execute' don't mean anything, as all the Nix store knows about is `execute`.

This is correct.

Reply:
> -Dm655 for everything else


`644` for everything else

![image](https://user-images.githubusercontent.com/9866621/188120563-ede10a33-5c61-4881-891c-dc53d3c1d1c9.png)

![image](https://user-images.githubusercontent.com/9866621/188120657-face5282-436b-41d3-a560-d8a7aba8e37d.png)

And as mentioned, post build it will be normalized to 0444 or 0555. if you use 755 and 655 every file will end up just 555 executable

Reply:
Yes, my mistake, I meant `644` for regular files.

Reply:
@azahi For this case, using `mv` is much simpler to read. And using `install` doesn't remove the `.so` files, it makes copies. This causes the automatic patchelf to get confused, and create references to `/build`, so I would need to `rm -rf *` in postInstall. All in all, it seems simpler to use `mv`.

Reply:
Alright, fair enough.